### DISCUSSION TITLE
Welcome to IdentityServer4.Admin Discussions!

### DISCUSSION THREAD

**Discussion by skoruba** [skoruba/IdentityServer4.Admin#765] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sfatz** [comment]
Hello, I've been trying to upgrade to the latest version of IdentityServer4.Admin (to match the latest version of IdentityServer4). However, I'm running into a small problem. I'm using the packages on NuGet, and one appears to be missing.  Skoruba.IdentityServer4.Shared. I can't seem to find it. As an alternative, I was thinking about just getting the source code, but it would be better to use a package in case there are future changes.

Is that missing from NuGet, or am I just searching wrong?

**Comment by sbn111** [comment]
[image](https://user-images.githubusercontent.com/59017140/113837704-bbbf2480-97ab-11eb-85f1-45cdbacfc29c.png)

Can you tell me why its showing like this. it showing while I am migrating to server

**Reply by skoruba** [reply]
Please, check the log for more detailed error message.

**Reply by sbn111** [reply]
Okay

**Comment by ognjenv** [comment]
Only with Identity.Admin certificate problem?

I passed procedure for docerize skoruba according manual, and identity.sts and identity.api (swagger) work perfect but identity.admin there is problem, I cannot open it through browser.

This is exception:

> [2022-12-05T20:03:26.3767024+00:00][INFO][79][Microsoft.AspNetCore.Hosting.Diagnostics] Request starting HTTP/1.1 GET http://admin.skoruba.local/ - -
> skoruba-identityserver4-admin         | [2022-12-05T20:03:26.3779765+00:00][INFO][79][Microsoft.AspNetCore.Authorization.DefaultAuthorizationService] Authorization failed. "These requirements were not met:\nRolesAuthorizationRequirement:User.IsInRole must be true for one of the following roles: (SkorubaIdentityAdminAdministrator)"
> skoruba-identityserver4-admin         | [2022-12-05T20:03:26.4907982+00:00][EROR][76][Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware] An unhandled exception has occurred while executing the request.
> skoruba-identityserver4-admin         | System.InvalidOperationException: IDX20803: Unable to obtain configuration from: 'System.String'.
> skoruba-identityserver4-admin         |  ---> System.IO.IOException: IDX20804: Unable to retrieve document from: 'System.String'.
> skoruba-identityserver4-admin         |  ---> System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
> skoruba-identityserver4-admin         |  ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid because of errors in the certificate chain: PartialChain
> skoruba-identityserver4-admin         |    at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
> skoruba-identityserver4-admin         |    at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
> skoruba-identityserver4-admin         |    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    --- End of inner exception stack trace ---
> skoruba-identityserver4-admin         |    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
> skoruba-identityserver4-admin         |    at Microsoft.IdentityModel.Protocols.HttpDocumentRetriever.GetDocumentAsync(String address, CancellationToken cancel)
> skoruba-identityserver4-admin         |    --- End of inner exception stack trace ---
> skoruba-identityserver4-admin         |    at Microsoft.IdentityModel.Protocols.HttpDocumentRetriever.GetDocumentAsync(String address, CancellationToken cancel)
> skoruba-identityserver4-admin         |    at Microsoft.IdentityModel.Protocols.OpenIdConnect.OpenIdConnectConfigurationRetriever.GetAsync(String address, IDocumentRetriever retriever, CancellationToken cancel)
> skoruba-identityserver4-admin         |    at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync(CancellationToken cancel)
> nginx                                 | nginx.1     | admin.skoruba.local 172.21.0.1 - - [05/Dec/2022:20:03:26 +0000] "GET / HTTP/2.0" 500 41616 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36" "172.21.0.7:80"
> skoruba-identityserver4-admin         |    --- End of inner exception stack trace ---
> skoruba-identityserver4-admin         |    at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync(CancellationToken cancel)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleChallengeAsyncInternal(AuthenticationProperties properties)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleChallengeAsync(AuthenticationProperties properties)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.ChallengeAsync(AuthenticationProperties properties)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authentication.AuthenticationService.ChallengeAsync(HttpContext context, String scheme, AuthenticationProperties properties)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at NWebsec.AspNetCore.Middleware.Middleware.CspMiddleware.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at NWebsec.AspNetCore.Middleware.Middleware.MiddlewareBase.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at NWebsec.AspNetCore.Middleware.Middleware.MiddlewareBase.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at NWebsec.AspNetCore.Middleware.Middleware.MiddlewareBase.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at NWebsec.AspNetCore.Middleware.Middleware.MiddlewareBase.Invoke(HttpContext context)
> skoruba-identityserver4-admin         |    at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
> skoruba-identityserver4-admin         | [2022-12-05T20:03:26.5155633+00:00][INFO][76][Microsoft.AspNetCore.Hosting.Diagnostics] Request finished HTTP/1.1 GET http://admin.skoruba.local/ - - - 500 - text/html;+charset=utf-8 138.8598ms
> 
> 


Can someone tell me where is problem with certificate with Identity.Admin?

If other two services work perfect, why Identiy.Admin has problem with certificate. Certificate assigned to *.skoruba.local, all subdomains are covered.
